bootstrap/helpers/parsers.php

```
@@ -514,84 +514,96 @@ function applicationParser(Application $resource, int $pull_request_id = 0, ?int
-                    $originalUrlFor = str($urlFor)->replace('_', '-');
-                    if (str($urlFor)->contains('-')) {
-                        $urlFor = str($urlFor)->replace('-', '_')->replace('.', '_');
+
+                    // Extract case-preserved service name from template
+                    $strKey = str($key->value());
+                    if ($parsed['has_port']) {
+                        if ($strKey->startsWith('SERVICE_URL_')) {
+                            $serviceNamePreserved = $strKey->after('SERVICE_URL_')->beforeLast('_')->value();
+                        } else {
+                            $serviceNamePreserved = $strKey->after('SERVICE_FQDN_')->beforeLast('_')->value();
+                        }
```

PHPMD (2.15.0)

The method applicationParser uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them. (undefined)

(ElseExpression)